engine/src/main/java/io/zeebe/engine/processor/AsyncSnapshotingDirectorService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Zeebe Workflow Engine
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.engine.processor;
+
+import io.zeebe.logstreams.log.LogStream;
+import io.zeebe.logstreams.state.StateSnapshotController;
+import io.zeebe.servicecontainer.Injector;
+import io.zeebe.servicecontainer.Service;
+import io.zeebe.servicecontainer.ServiceStartContext;
+import io.zeebe.servicecontainer.ServiceStopContext;
+import java.time.Duration;
+
+public class AsyncSnapshotingDirectorService implements Service<AsyncSnapshotingDirectorService> {
+  private final Injector<StreamProcessorService> streamProcessorServiceInjector = new Injector<>();
```

Comment:
I tried it but I have to partially initialize the Director on creation and then set the stream processor controller things on service start. That makes it harder to test (we can make the stream process service and inject it manually but still it had some errors that didn't seem obvious) and it wasn't a very elegant solution to have the initialization split because of the dependencies. But I agree that in services without injectors (like the LogBlockIndexService) it would be a nice refactor